src/Aspire.Hosting/Dashboard/DashboardLifecycleHook.cs

```diff
@@ -160,6 +161,32 @@ private void ConfigureAspireDashboardResource(IResource dashboardResource)
             if (otlpHttpEndpointUrl != null)
             {
                 context.EnvironmentVariables[DashboardConfigNames.DashboardOtlpHttpUrlName.EnvVarName] = otlpHttpEndpointUrl;
+
+                var model = context.ExecutionContext.ServiceProvider.GetRequiredService<DistributedApplicationModel>();
+                var allResourceEndpoints = model.Resources
+                    .Where(r => !string.Equals(r.Name, KnownResourceNames.AspireDashboard, StringComparisons.ResourceName))
+                    .SelectMany(r => r.Annotations)
+                    .OfType<EndpointAnnotation>()
+                    .ToList();
+
+                var corsOrigins = new HashSet<string>();
+                foreach (var endpoint in allResourceEndpoints)
+                {
+                    if (endpoint.UriScheme is "http" or "https")
+                    {
+                        if (endpoint.Port != null)
+                        {
+                            corsOrigins.Add($"{endpoint.UriScheme}://localhost:{endpoint.Port}");
+                        }
+                        if (endpoint.TargetPort != null)
+                        {
+                            corsOrigins.Add($"{endpoint.UriScheme}://localhost:{endpoint.TargetPort}");
+                        }
+                    }
+                }
```

Comment:
@mitchdenny Is this the right way to get all of the resource endpoints? I noticed that sometimes the `EndpointAnnotation.Port` value was missing or wrong.

For example, one endpoint in TestShop says it has a port of 80, which is wrong. But it has an `EndpointAnnotation.AllocatedEndpoint` which looks like it reports the correct port.

Reply:
Yeah allocated endpoint is what you want.

Reply:
Yeah, if you want the actual ports as it was determined by DCP then you want the allocated endpoint. Docs can definitely be improved.

If the Port on the EndpointAnnotation is null you are basically saying that you want DCP to allocate a random one.